File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int missing = -1, repeating = -1;
        int n = grid.length;
        int size = n * n;
        int[] freq = new int[size + 1];

        for (int[] row : grid) {
            for (int num : row) {
                freq[num]++;
            }
        }

        for (int i = 1; i <= size; i++) {
            if (freq[i] == 2) {
                repeating = i;
            }
            if (freq[i] == 0) {
                missing = i;
            }
        }
        
        return new int[]{repeating, missing};
    }
}

File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& grid) {
        int missing, repeating;
        int n=grid[0].size();
        int size=n*n;
        vector<int> freq(size+1,0);
        for (int i = 0; i < grid.size(); i++) {
            for (int j = 0; j < grid[i].size(); j++) {
                freq[grid[i][j]]++;
            }
        }
        for(int i=1;i<=size;i++){
            if(freq[i]==2){
                repeating=i;
            }
            if(freq[i]==0){
                missing=i;
            }
        }
        return {repeating, missing};
    }
};